### PR TITLE
Installer

### PR DESCRIPTION
**Pull request by kevinprince**
- add install and update script
- add helpers for setting up gfio-profiles
- add helpers for aws / docker cred linking
- removed rancher related tooling
- upgraded docker compose

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,46 +5,136 @@ This repo is made up of two parts:
...
+*If you have an existing install, move your .gfio folder to another place and
+copy you existing profiles back once installation is complete.*
 
-**Profile switching**: When `gfio` is called, it walks up the directory tree
+- Run AWS config
+
+```shell
+  cd ~
+  gfio aws configure
+```
```
Comment (avit) [inline README.md]: Users will need gfio in PATH for this to work immediately after install. Maybe mention it; we probably shouldn't make the installer change your bash config files.

File: README.md [hunk 2]
```diff
@@ -5,46 +5,136 @@ This repo is made up of two parts:
...
+    gfio machine-create-aws 10.web.production.gfio.ca
+    AWS_INSTANCE_TYPE=c4.large gfio machine-create-aws 99.bigbox.production.gfio.ca
+  ```
+
+*Rackspace*
+
+```shell
+  gfio machine-create-rs 10.web.production.gfio.ca
+  OS_FLAVOR_ID=general1-2 gfio machine-create-rs 99.bigbox.production.gfio.ca
+```
```
Comment (avit) [inline README.md]: Mention default instance types here?

File: rootfs/usr/local/bin/profile-set [hunk 3]
```diff
@@ -0,0 +1 @@
+echo $1 >> $PWD/.gfio_profile
```
Comment (avit) [inline rootfs/usr/local/bin/profile-set]: Check for $1 or abort: don't clear profile

File: updater.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+echo ""
+
+if [[ ! -d $HOME/.gfio/src ]]; then
+  echo "----> The src directory is missing please re-install gfio-cli"
+  exit 1
+fi
+
+cd $HOME/.gfio/src
+git checkout master
+git reset --hard
```
Comment (avit) [inline updater.sh]: Abort if local copy has changes?

File: Dockerfile [hunk 5]
```diff
@@ -29,12 +29,6 @@ RUN bash -c "\
   unzip -qq -d ${BIN_PATH} terraform_${TERRAFORM_VERSION}_linux_amd64.zip ;\
   rm terraform_${TERRAFORM_VERSION}_linux_amd64.zip"
 
-ENV RANCHER_VERSION=0.7.4
-RUN bash -c "\
-  curl -sSL https://github.com/rancher/rancher-compose/releases/download/v${RANCHER_VERSION}/rancher-compose-linux-amd64-v${RANCHER_VERSION}.tar.gz | tar -zxf - ;\
-  mv rancher-compose-v${RANCHER_VERSION}/rancher-compose ${BIN_PATH} ;\
-  rm -r rancher-compose-v${RANCHER_VERSION}"
-
```
Comment (iamveen) [inline Dockerfile]: Nooooooooo

File: install.sh [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+docker -v docker >/dev/null 2>&1 || {
+  echo >&2 "----> Docker is required but it's not installed.  Aborting."; exit 1;
+}
+
+if [ -z "$DOCKER_HOST" ]; then
+    echo "----> Docker is installed but not accessable is it running?"
+    exit 1
+fi
```
Comment (iamveen) [inline install.sh]: This is not compatible with Docker for Mac, as it doesn't use ENV vars the same way docker-machine does.

Comment (iamveen) [inline install.sh]:
```
ls -l /var/run/docker.sock
lrwxr-xr-x  1 root  daemon  60 17 May 15:28 /var/run/docker.sock -> /Users/iamveen/Library/Containers/com.docker.docker/Data/s60
```

**Comment by iamveen** [comment]
Good practice for those bash scripts would be:

```
#!/usr/bin/env bash

set -o errexit
set -o pipefail
set -o nounset
```

I'm curious how they work at all without the interpreter being set, didn't know that was possible,

**Pull request merged**